### ISSUE TITLE
Use xmake instead of cmake

### ISSUE THREAD

**Issue by darcy-shen** [galkahana/PDF-Writer#203]
See https://github.com/XmacsLabs/mogan/pull/544

We are migrating Mogan from cmake to xmake.

Using xmake, we do not need to include the source code of the Hummus PDF Writer.

Just:
```
add_requires("pdfhummus 4.1",{system=false,configs={libpng=true,libjpeg=true}})
```

Here is the package definition for pdfhummus contributed by @jingkaimori.

https://github.com/xmake-io/xmake-repo/blob/master/packages/p/pdfhummus/xmake.lua

**Comment by galkahana** [comment]
you can also avoid including pdfhummus with cmake, in case that's interesting for you:
https://github.com/galkahana/pdf-text-extraction/blob/master/CMakeLists.txt#L10

**Comment by jingkaimori** [comment]
> would adding xmake.lua file with this content be helpful?

Yes, but this file does not contain tests, and cannot use bundled dependencies. Some extra works may be necessary.

**Comment by darcy-shen** [comment]
https://github.com/xmake-io/xmake/issues/3381

xmake test will be available in xmake 2.8.4, we can enable unit tests when xmake 2.8.4 is avaible.

Just created pull request to add xmake support: https://github.com/galkahana/PDF-Writer/pull/228

**Comment by galkahana** [comment]
cheers. i'll take it. just to be sure. im not doing this, and will not do this, in replacement of cmake. and i will continue to support bundled and unbundled versions for cmake.

**Comment by darcy-shen** [comment]
>  im not doing this, and will not do this

It is reasonable. CMake is much more popular than xmake for now. Let me maintain the xmake.lua and we (XmacsLabs) will maintain it.

**Comment by galkahana** [comment]
truly appreciate your contribution
